### PR TITLE
Get rid of empty values from default-config output

### LINKED ISSUE #658: Output of default-config can not be parsed by k0s
```
# k0s version
v0.10.0-beta2
# k0s default-config
```

==> 

```yaml
apiVersion: k0s.k0sproject.io/v1beta1
images:  # <-- is this supposed to be on top level?
# ...
spec:
# ...
  storage:
    kine: null # <-- breaks yaml unmarshal
# ...
```

### PR DESCRIPTION
**Pull request by jnummelin**
Signed-off-by: Jussi Nummelin <jnummelin@mirantis.com>

**Issue**
Fixes #658 #520 

**What this PR Includes**
Gets rid of the empty values from `k0s default-config` output. After the changes:
```
$ ./k0s default-config
apiVersion: k0s.k0sproject.io/v1beta1
images:
  konnectivity:
    image: us.gcr.io/k8s-artifacts-prod/kas-network-proxy/proxy-agent
    version: v0.0.13
  metricsserver:
    image: gcr.io/k8s-staging-metrics-server/metrics-server
    version: v0.3.7
  kubeproxy:
    image: k8s.gcr.io/kube-proxy
    version: v1.20.2
  coredns:
    image: docker.io/coredns/coredns
    version: 1.7.0
  calico:
    cni:
      image: calico/cni
      version: v3.16.2
    flexvolume:
      image: calico/pod2daemon-flexvol
      version: v3.16.2
    node:
      image: calico/node
      version: v3.16.2
    kubecontrollers:
      image: calico/kube-controllers
      version: v3.16.2
installConfig:
  users:
    etcdUser: etcd
    kineUser: kube-apiserver
    konnectivityUser: konnectivity-server
    kubeAPIserverUser: kube-apiserver
    kubeSchedulerUser: kube-scheduler
kind: Cluster
metadata:
  name: k0s
spec:
  api:
    address: 172.31.36.24
    sans:
    - 172.31.36.24
    - 172.17.0.1
    - 172.24.0.1
    - 172.25.0.1
    - 172.26.0.1
    - 172.19.0.1
    - 172.27.0.1
    - 172.18.0.1
  storage:
    type: etcd
    etcd:
      peerAddress: 172.31.36.24
  network:
    podCIDR: 10.244.0.0/16
    serviceCIDR: 10.96.0.0/12
    provider: calico
    calico:
      mode: vxlan
      vxlanPort: 4789
      vxlanVNI: 4096
      mtu: 1450
      wireguard: false
      flexVolumeDriverPath: /usr/libexec/k0s/kubelet-plugins/volume/exec/nodeagent~uds
      withWindowsNodes: false
      overlay: Always
  podSecurityPolicy:
    defaultPolicy: 00-k0s-privileged
telemetry:
  interval: 10m0s
  enabled: true
```

### CODE REVIEW

**Review by jasmingacic — APPROVED** [review]

File: pkg/apis/v1beta1/cluster.go [hunk 1]
```diff
@@ -175,20 +185,27 @@ func DefaultAPISpec() *APISpec {
 	// Collect all nodes addresses for sans
 	addresses, _ := util.AllAddresses()
 	publicAddress, _ := util.FirstPublicAddress()
+	addresses = append(addresses, publicAddress)
+	addresses = util.Unique(addresses)
```
Comment (ncopa) [inline pkg/apis/v1beta1/cluster.go:189]: This is the fix for #520, right? Might be nice to have in a separate commit since it does not have anything to do with empty values.

File: pkg/apis/v1beta1/cluster.go [hunk 2]
```diff
@@ -175,20 +185,27 @@ func DefaultAPISpec() *APISpec {
 	// Collect all nodes addresses for sans
 	addresses, _ := util.AllAddresses()
 	publicAddress, _ := util.FirstPublicAddress()
+	addresses = append(addresses, publicAddress)
```
Comment (ncopa) [inline pkg/apis/v1beta1/cluster.go:188]: why do we need to add first public address to `AllAddresses`?

Comment (jasmingacic) [reply pkg/apis/v1beta1/cluster.go:188]: I believe it was there before Jussi only moved it up and filtered it with util.Unique

Comment (jnummelin) [reply pkg/apis/v1beta1/cluster.go:188]: uh, we don't. `AllAddresses()` should give us all addresses already. I'll fix this

**Review by ncopa — APPROVED** [review]
